### DISCUSSION TITLE
if you open local scores while global scores are loading, global scores load instead

### DISCUSSION THREAD

**Discussion by chayleaf** [ppy/osu#15617] [Q&A] [answered]
It happens only if you get the timing exactly right, here's a video of me trying to reproduce it and only succeeding at the very end (4:10): https://youtu.be/I5iqiAfYoiw

I didn't react quickly enough and switched away from it, but it still got caught on video.

In this case, global "personal best" is shown at the bottom on the local leaderboard. I'm pretty sure I saw the entire global leaderboard being shown to me instead of the local leaderboard once, but I didn't screenshot it.

**Comment by bdach** [comment]
if you mean that "global scores load" on this screen:

![2021-11-14-121532_886x655_scrot](https://user-images.githubusercontent.com/20418176/141678607-ed80c3f6-f622-4398-a464-8c359a3ab225.png)

then that's not a "bug", that's how it was written.

https://github.com/ppy/osu/blob/91a21152dff2ac77b45450edfb73692d2860a13d/osu.Game/Screens/Ranking/SoloResultsScreen.cs#L28-L36

unless you want the results screen to only show local scores if you click to it via the "local" tab. which would be a feature request.

**Reply by chayleaf** [reply]
no, i mean the leaderboard on the song selection screen, i clicked on the score on accident, ignore that

**Reply by bdach** [reply]
do you mean this state, then?

![2021-11-14-152218_586x303_scrot](https://user-images.githubusercontent.com/20418176/141685165-fcde222e-3631-4a94-9efd-842831aafc13.png)

the leaderboard is still correct in a sense, it's just that the user top score should not be showing.

**Reply by chayleaf** [reply]
yeah, which is a bug nonetheless

as I mentioned, I'm pretty sure the entire leaderboard might get overwritten too with correct timing, but that doesn't matter as "your personal best" being shown here is a bug as well

**Comment by bdach** [comment, marked as answer]
escalated to issue #15622
